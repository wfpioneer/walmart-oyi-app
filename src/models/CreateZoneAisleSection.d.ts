import { CreateAisles } from '../state/reducers/Location';

export interface CreateAisleRequest {
  zoneId: number;
  aisles: CreateAisles[]
}

export interface ZoneAislesSectionResponse {
  aisleId: number;
  aisleName: number;
  createdSectionsResponseDto: CreateAisleSectionsResponse;
  sectionCount: number;
  status: number;
}

export interface CreateZoneAisleSectionResponse {
  aisles: ZoneAislesSectionResponse[];
  status: number;
  zoneId: number;
  zoneName: string;
}

export interface CreateAisleResponse {
  aisleId: number;
  aisleName: number;
  status: number;
  message: string;
  createdSectionsResponseDto: CreateAisleSectionsResponse;
}

interface CreateAisleSectionsResponse {
  aisleId: number;
  status: number;
  message: string;
  createdSections: string[]
}
