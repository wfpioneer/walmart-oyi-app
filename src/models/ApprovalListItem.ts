export interface ApprovalListItem {
  imageUrl?: string;
  itemName: string;
  itemNbr: number;
  upcNbr: number;
  categoryNbr: number;
  categoryDescription: string;
  subCategoryNbr: number;
  subCategoryDescription: string;
  oldQuantity: number;
  newQuantity: number;
  dollarChange: number;
  initiatedUserId: string;
  initiatedTimestamp: string;
  approvalStatus: approvalStatus;
  approvalRequestSource: approvalRequestSource;
  isChecked?: boolean;
  daysLeft?: number;
}

export interface ApprovalCategory extends ApprovalListItem {
  categoryHeader?: boolean;
}

export enum approvalStatus {
  'Pending' = 'Pending',
  'Approved' = 'Approved',
  'Rejected' = 'Rejected',
  'Expired' = 'Expired'
}
export enum approvalRequestSource {
  'ItemDetails' = 'itemdetails',
  'Audits' = 'audits',

}
