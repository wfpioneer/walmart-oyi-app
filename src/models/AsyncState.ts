export interface AsyncState{
  isWaiting: boolean;
  value: any;
  error: any;
  result: any;
}
