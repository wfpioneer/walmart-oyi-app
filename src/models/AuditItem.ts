export interface ItemPalletInfo {
  palletId: string,
  quantity: number,
  sectionId: number,
  locationName: string,
  mixedPallet: boolean,
  newQty?: number
}
