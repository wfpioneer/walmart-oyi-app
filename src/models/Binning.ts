export interface BinningItem {
  itemNbr: number;
  itemDesc: string;
  price: number;
  upcNbr: string;
  quantity: number;
}
export interface BinningPallet {
  id: number;
  expirationDate: string;
  lastLocation?: string;
  firstItem: BinningItem;
}