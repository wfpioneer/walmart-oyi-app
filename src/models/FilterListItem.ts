interface FilterListItem {
    value: string;
    display: string;
    selected: boolean;
  }

export default FilterListItem;
