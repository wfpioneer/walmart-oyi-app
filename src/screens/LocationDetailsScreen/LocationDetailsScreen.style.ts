import { StyleSheet } from 'react-native';
import { COLOR } from '../../themes/Color';

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  item: {
    backgroundColor: COLOR.WHITE,
    padding: 14,
    marginVertical: 1,
    marginHorizontal: 0
  },
  title: {
    fontSize: 40
  },
  itemContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center'
  },
  noAisles: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  activityIndicator: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  errorView: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 15,
    paddingVertical: 15
  },
  errorButton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLOR.RED_300,
    width: '95%',
    borderRadius: 4,
    height: 40,
    marginVertical: 10
  },
  errorText: {
    fontSize: 16,
    textAlign: 'center'
  },
  contentPadding: {
    paddingBottom: 100
  },
  tabHeader: {
    backgroundColor: COLOR.WHITE,
    padding: 14,
    marginHorizontal: 0,
    borderBottomWidth: 2,
    borderBottomColor: COLOR.GREY_200,
    flexDirection: 'row'
  },
  clear: {
    color: COLOR.MAIN_THEME_COLOR,
    paddingRight: 14,
    fontSize: 14
  },
  add: {
    color: COLOR.MAIN_THEME_COLOR,
    paddingLeft: 14,
    fontSize: 14
  },
  tabHeaderText: {
    flex: 1,
    color: COLOR.GREY_600,
    fontSize: 14
  },
  pipe: {
    color: COLOR.GREY_600,
    fontSize: 14
  }
});

export default styles;
