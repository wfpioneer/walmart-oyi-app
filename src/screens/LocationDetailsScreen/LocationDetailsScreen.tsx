import React, { Dispatch, EffectCallback, useEffect } from 'react';
import {
  ActivityIndicator, Text, TouchableOpacity, View
} from 'react-native';
import {
  NavigationProp, RouteProp, useNavigation, useRoute
} from '@react-navigation/native';
import { useDispatch } from 'react-redux';
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import { FlatList } from 'react-native-gesture-handler';
import { useTypedSelector } from '../../state/reducers/RootReducer';
import { strings } from '../../locales';
import { LocationItem } from '../../models/LocationItems';
import { getSectionDetails } from '../../state/actions/saga';
import { AsyncState } from '../../models/AsyncState';
import styles from './LocationDetailsScreen.style';
import COLOR from '../../themes/Color';
import { trackEvent } from '../../utils/AppCenterTool';

interface LocationDetailProps {
  zoneName: string;
  aisleName: string;
  sectionName: string;
  getSectionDetailsApi: AsyncState;
  dispatch: Dispatch<any>;
  navigation: NavigationProp<any>;
  route: RouteProp<any, string>;
  trackEventCall: (eventName: string, params?: any) => void;
  useEffectHook: (effect: EffectCallback, deps?:ReadonlyArray<any>) => void;
}

export const LocationDetailsScreen = (props: LocationDetailProps) : JSX.Element => {
  const {
    zoneName,
    aisleName,
    sectionName,
    getSectionDetailsApi,
    navigation,
    route,
    dispatch,
    trackEventCall,
    useEffectHook
  } = props;

  const locationItem: LocationItem | undefined = (getSectionDetailsApi.result && getSectionDetailsApi.result.data);

  if (getSectionDetailsApi.isWaiting) {
    return (
      <ActivityIndicator
        animating={getSectionDetailsApi.isWaiting}
        hidesWhenStopped
        color={COLOR.MAIN_THEME_COLOR}
        size="large"
        style={styles.activityIndicator}
      />
    );
  }

  if (getSectionDetailsApi.error) {
    return (
      <View style={styles.errorView}>
        <MaterialCommunityIcon name="alert" size={40} color={COLOR.RED_300} />
        <Text style={styles.errorText}>{strings('LOCATION.LOCATION_API_ERROR')}</Text>
        <TouchableOpacity
          style={styles.errorButton}
          onPress={() => {
            trackEventCall('location_api_retry',);
            dispatch(getSectionDetails({ sectionId: `${zoneName}${aisleName}-${sectionName}` }));
          }}
        >
          <Text>{strings('GENERICS.RETRY')}</Text>
        </TouchableOpacity>
      </View>
    );
  }
  return (
    <View style={styles.locDetailsScreenContainer}>
      <Text> PlaceHolder for the List</Text>
    </View>
  );
};

const LocationDetails = (): JSX.Element => {
  const sectionName = useTypedSelector(state => state.Location.selectedSection.name);
  const zoneName = useTypedSelector(state => state.Location.selectedZone.name);
  const aisleName = useTypedSelector(state => state.Location.selectedAisle.name);
  const getSectionDetailsApi = useTypedSelector(state => state.async.getSectionDetails);
  const navigation = useNavigation();
  const route = useRoute();
  const dispatch = useDispatch();
  return (
    <>
      <LocationDetailsScreen
        zoneName={zoneName}
        aisleName={aisleName}
        sectionName={sectionName}
        getSectionDetailsApi={getSectionDetailsApi}
        dispatch={dispatch}
        navigation={navigation}
        route={route}
        trackEventCall={trackEvent}
        useEffectHook={useEffect}
      />
    </>
  );
};

export default LocationDetails;
