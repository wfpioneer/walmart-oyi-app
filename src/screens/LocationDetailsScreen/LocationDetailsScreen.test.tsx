import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { NavigationProp, Route } from '@react-navigation/native';
import { LocationDetailsScreen } from './LocationDetailsScreen';
import { AsyncState } from '../../models/AsyncState';
import {
  mockLocationDetails,
  mockLocationDetailsEmpty,
  mockLocationDetailsLargeLocationCount
} from '../../mockData/locationDetails';

let navigationProp: NavigationProp<any>;
let routeProp: Route<any>;

const SECTION_NAME = '1';
const ZONE_NAME = 'ABAR';
const AISLE_NAME = '2';

// TODO Adjust Snapshot naming convention Remove on final PR
describe('Test Location Details Screen', () => {
  const defaultAsyncState: AsyncState = {
    isWaiting: false,
    value: null,
    error: null,
    result: null
  };
  it('Renders Location Details Screen with Mock Data', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <LocationDetailsScreen
        sectionName={SECTION_NAME}
        zoneName={ZONE_NAME}
        aisleName={AISLE_NAME}
        mockData={mockLocationDetails}
        getSectionDetailsApi={defaultAsyncState}
        dispatch={jest.fn()}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('Test Location Details Screen', () => {
  const defaultAsyncState: AsyncState = {
    isWaiting: false,
    value: null,
    error: null,
    result: null
  };
  it('Renders Location Details Screen with Mock Empty Data', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <LocationDetailsScreen
        sectionName={SECTION_NAME}
        zoneName={ZONE_NAME}
        aisleName={AISLE_NAME}
        mockData={mockLocationDetailsEmpty}
        getSectionDetailsApi={defaultAsyncState}
        dispatch={jest.fn()}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('Test Location Details Screen', () => {
  const defaultAsyncState: AsyncState = {
    isWaiting: false,
    value: null,
    error: null,
    result: null
  };
  it('Renders Location Details Screen with Mock Large Location Data', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <LocationDetailsScreen
        sectionName={SECTION_NAME}
        zoneName={ZONE_NAME}
        aisleName={AISLE_NAME}
        mockData={mockLocationDetailsLargeLocationCount}
        getSectionDetailsApi={defaultAsyncState}
        dispatch={jest.fn()}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
