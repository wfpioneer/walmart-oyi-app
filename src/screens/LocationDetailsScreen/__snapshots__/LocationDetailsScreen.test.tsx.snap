// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Location Details Screen Renders Location Details Screen with Mock Data 1`] = `
<View>
  <LocationHeader
    details="2 [missing \\"en.LOCATION.ITEMS\\" translation], 3 [missing \\"en.LOCATION.PALLETS\\" translation]"
    location="[missing \\"en.LOCATION.SECTION\\" translation] ABAR2-1"
  />
</View>
`;

exports[`Test Location Details Screen Renders Location Details Screen with Mock Empty Data 1`] = `
<View>
  <LocationHeader
    details="0 [missing \\"en.LOCATION.ITEMS\\" translation], 0 [missing \\"en.LOCATION.PALLETS\\" translation]"
    location="[missing \\"en.LOCATION.SECTION\\" translation] ABAR2-1"
  />
</View>
`;

exports[`Test Location Details Screen Renders Location Details Screen with Mock Large Location Data 1`] = `
<View>
  <LocationHeader
    details="100 [missing \\"en.LOCATION.ITEMS\\" translation], 100 [missing \\"en.LOCATION.PALLETS\\" translation]"
    location="[missing \\"en.LOCATION.SECTION\\" translation] ABAR2-1"
  />
</View>
`;
